src/services/base/base.service.js

```diff
@@ -866,49 +875,70 @@ module.exports.search = async (args, {req}, resource_name, collection_name) => {
              * @type {string[]}
              */
             const sort_properties_list = sort_properties_as_csv.split(',');
-            for (let i in sort_properties_list) {
-                // noinspection JSUnfilteredForInLoop
+            if (sort_properties_list.length > 0) {
+                /**
+                 * @type {import('mongodb').Sort}
+                 */
+                const sort = {};
                 /**
                  * @type {string}
                  */
-                const x = sort_properties_list[`${i}`];
-                if (x.startsWith('-')) {
-                    // eslint-disable-next-line no-unused-vars
-                    /**
-                     * @type {string}
-                     */
-                    const x1 = x.substring(1);
-                    cursor = cursor.sort({[x1]: -1});
-                } else {
-                    cursor = cursor.sort({[x]: 1});
+                for (const sortProperty of sort_properties_list) {
+                    if (sortProperty.startsWith('-')) {
+                        /**
+                         * @type {string}
+                         */
+                        const sortPropertyWithoutMinus = sortProperty.substring(1);
+                        sort[`${sortPropertyWithoutMinus}`] = -1;
+                    } else {
+                        sort[`${sortProperty}`] = 1;
+                    }
                 }
+                options['sort'] = sort;
             }
         }
 
+        // if _count is specified then limit mongo query to that
         if (combined_args['_count']) {
             // for consistency in results while paging, always sort by _id
             // https://docs.mongodb.com/manual/reference/method/cursor.sort/#sort-cursor-consistent-sorting
-            cursor = cursor.sort({'_id': 1});
+            options['sort'] = {'_id': 1};
```

Comment:
@imranq2 While testing EOB endpoint, I found that when count and sort params are given, the input sort param is overridden by the id sort whereas the expectation would be that the input sort param gets applied as well.

Reply:
Fixed in another PR.  thanks